### PR TITLE
Feature/11/testfile discovery

### PR DESCRIPTION
**Pull request by julianxyz**

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -1,46 +1,34 @@
...
+            files = listDir(args[0], true);
         } else {
-            console.error(chalk.red(`${element} is a directory`));
+            files = getCanonicalPaths(args);
         }
-    } catch (e) {
-        console.error(chalk.red(`file ${element} does not exist`));
-    }
-});
+    });
```
Comment (s1hofmann) [inline cli.js:30]: We are doing duplicated work here when assigning `files` with the list of test files.

This part should be split into filtering / checking for the `--recursive` flag and actually parsing a list of paths.

File: testrunner.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                tempfiles = tempfiles.concat(listDir(element));
+            } else {
+                tempfiles.push(element);
+            }
+        } catch (e) {
+            console.error(chalk.red(`file ${element} does not exist`));
+        }
+    });
+    return tempfiles;
+}
```
Comment (s1hofmann) [inline testrunner.js:22]: This function looks quite similar to `listDir`. Any chance we could re-use something?

File: listDir.js [hunk 3]
```diff
@@ -3,20 +3,35 @@ const { join } = require("path");
...
+                }
             }
-        }
-    });
-    return paths;
+        }).reduce((akk, elem) => akk.concat(elem), [])
+          .filter(elem => elem !== undefined);
+    } else {
+        return [path];
+    }
```
Comment (s1hofmann) [inline listDir.js]: This recursive call could be simplified!

File: cli.js [hunk 4]
```diff
@@ -11,24 +10,24 @@ let [, , ...args] = process.argv;
...
+// Checking for recursion flag.
+let isRecursive = false;
+const inputpaths = args.filter((elem) => {
+    if (elem === '--recursive'){
+        isRecursive = true;
+        return false;
+    } else {
+        return true;
+    }
+});
```
[the remark this answers is not in the document]
Comment (julianxyz) [reply cli.js]: Should be ok now.

**Pull request merged**